compiler/compile-config/compile-config-2.01-grounding

```diff
@@ -0,0 +1,739 @@
+            [[ ! -e \"$DEEPDIVE_LOGFILE\" ]] || mv -f \"$DEEPDIVE_LOGFILE\" \"$DEEPDIVE_LOGFILE\"~
+            java org.deepdive.Main -c <(
+                set +x
+                echo \("deepdive \(.deepdive | @json)" | @sh)
+                echo \("deepdive.pipeline.pipelines.grounding: [\($deepdive.inference.factors_ | map(.factorName) | join(", "))]" | @sh)
+                echo \("deepdive.pipeline.run: grounding" | @sh)
+            ) -o factorgraph -t inference_grounding
+
+            # drop graph. prefix from file names
+            cd factorgraph
+            mv -f graph.variables variables
+            mv -f graph.factors   factors
+            mv -f graph.weights   weights
+            mv -f graph.meta      meta
+        "
+    }
+}
+
+
+###############################################################################
+
+## variable_id_partition
+# Grounding begins by counting the variables to partition a range of
+# non-negative integers for assigning the variable ids.
+| .deepdive_.execution.processes += {
+    "process/grounding/variable_id_partition": {
+        dependencies_: [
+            # id partition depends on all variable tables
+            $deepdive.schema.variables_[] | "data/\(.variablesTable)"
+        ],
+        style: "cmd_extractor", cmd: "
+        : ${DEEPDIVE_GROUNDING_DIR:=\"$DEEPDIVE_APP\"/run/model/grounding}
+
+        RANGE_BEGIN=0 \\
+        partition_id_range \($deepdive.schema.variables_ | map(.variablesTable | @sh) | join(" ")) | {
+            # record the base
+            variableCountTotal=0
+            while read table begin excludeEnd; do
+                varPath=\"$DEEPDIVE_GROUNDING_DIR\"/variable/${table}
+                mkdir -p \"$varPath\"
+                cd \"$varPath\"
+                echo $begin                      >id_begin
+                echo $excludeEnd                 >id_exclude_end
+                echo $(( $excludeEnd - $begin )) >count
+                variableCountTotal=$excludeEnd
+            done
+            # record the final count
+            echo $variableCountTotal >\"$DEEPDIVE_GROUNDING_DIR\"/variable_count
+        }
+        "
+    }
+}
+
+
+## variable/*/assign_id
+# Each variable table then gets the range of integers assigned to the id column
+# of every row.
+| .deepdive_.execution.processes += merge($deepdive.schema.variables_[] | {
+    "process/grounding/variable/\(.variableName)/assign_id": {
+        dependencies_: [
+            "process/grounding/variable_id_partition"
+        ],
+        style: "cmd_extractor", cmd: "
+        : ${DEEPDIVE_GROUNDING_DIR:=\"$DEEPDIVE_APP\"/run/model/grounding}
+        table=\(.variablesTable | @sh)
+
+        cd \"$DEEPDIVE_GROUNDING_DIR\"/variable/${table}
+        baseId=$(cat id_begin)
+
+        # assign id to all rows according to the paritition
+        deepdive db assign_sequential_id $table \($deepdiveVariableIdColumn | @sh) $baseId
+
+        \(
+        if .variableType == "categorical" then
```

Comment:
I agree.  Maybe we should rename this step to a more general one?  Any good idea?  Alternatively, we could create optional process nodes, but I didn't want to mess up with the dependency graph as it requires another branch when enumerating dependencies for the `"process/grounding/variable_holdout"` node.